resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java

```diff
@@ -51,31 +53,25 @@
    private final List<Consumer<Throwable>> onErrorConsumers = new CopyOnWriteArrayList<>();
 
    private final List<Runnable> onCompleteConsumers = new CopyOnWriteArrayList<>();
+   
+   // No need to be volatile since visibility is guarantees by the fact that it will always be set before setting 
+   // AtomicReference state to 'OPEN' and read only if AtomicReference state is set to 'OPEN'.
```

Comment:
Is this for AtomicReference<State> ?

Reply:
This comment is just to say that I voluntarily  did not make `response` field volatile.
Its visibility in `close(...)` method will be ensured by the happen before relationship established by the `AtomicReference state` field.

In other words, if state == Open is true in close method, then thanks to happen before transitivity, reading of `response` field will read the most up to date version of that field.

This is ensured by the fact that we always write to `response` field before setting `state` field
to OPEN:

```
      @Override
      public void run()
      {
           ...
            // response must be set before switching to OPEN state to ensure that it will be visible
            // in close method.
            response = localResponse;
            // Let's switch state from CONNECTING to OPEN if possible and process event stream.
            if (state.compareAndSet(State.CONNECTING, State.OPEN))
           ...
        }
```

So in close method If I can see that `AtomicReference state` is OPEN I will also see any writes that happen prior to setting `AtomicReference state` to OPEN.